src/SectorView.cpp

```diff
@@ -507,6 +731,65 @@ void SectorView::Draw3D()
 		}
 		DrawRouteLines(modelview);
 	}
+}
+
+// function is used to pass label parameters from lua
+void SectorView::SetLabelParams(std::string fontName, int fontSize, float gap, Color highlight, Color shade)
+{
+	m_labels.fontName = fontName;
+	m_labels.fontSize = fontSize;
+	m_labels.gap = gap;
+	m_labels.colors.highlight = IM_COL32(highlight.r, highlight.g, highlight.b, highlight.a);
+	m_labels.colors.shade = IM_COL32(shade.r, shade.g, shade.b, shade.a);
+}
+
+// function runs from lua, inside imgui context
+void SectorView::DrawLabels()
```

Comment:
Instead of Lua calling into this function, prefer calling this from c++ under the `DrawPiGui` virtual function and let Lua call in to only enable/disable drawing of labels. This is mostly for issues with Lua function continuations and to allow us to further harden against Lua errors - as much as possible should be event driven and I want to avoid large, lengthy functions in C++ executing inside of a lua callstack context if possible.

Additionally, there's room to go the extra mile and convert the SectorMap to run its rendering on the appropriate handler instead of the `game` pigui handler - this now works for all PiGuiViews.

Reply:
Done!

Reply:
Looking good, my only suggestion is to allow pigui to control whether or not labels are being drawn by setting a variable (to fully replicate previous behavior)

Reply:
Done!